core/wiki/macros/translink.tid

```diff
@@ -3,14 +3,25 @@ tags: $:/tags/Macro
 
 \define translink(title,mode:"block")
 \whitespace trim
-<div style="border:1px solid #ccc; padding: 0.5em; background: black; foreground; white;">
+<div class={{{ [[tc-translink]] [[$title$]is[missing]then[tc-translink-inline]] [[$mode$]match[block]then[tc-translink-block]] [[$mode$]match[inline]then[tc-translink-inline]] +[first[2]] +[join[ ]] }}}>
```

Comment:
We avoid using the textual substitution feature of macros unless absolutely necessary. In this case, using textual substitution means that tiddler titles containing `]]` will not behave correctly.

The alternative is to use the double underscore syntax to access the parameters as variables:

```
 <div class={{{ [[tc-translink]] [<__title__>]is[missing]then[tc-translink-inline]] [<__mode__>match[block]then[tc-translink-block]] [<__mode__>match[inline]then[tc-translink-inline]] +[first[2]] +[join[ ]] }}}>
```